### DISCUSSION TITLE
DB questions.

### DISCUSSION THREAD

**Discussion by xGuysOG** [basecamp/kamal#519] [Q&A]
i have 2 rather simple questions about kamal:
1. Can you deploy without the need for a DB?, just a normal site deploy?
2. Are you forced to work with MySQL or can you move it around and use Mogo, sqlLite og postgres?

**Comment by SadeeshaJayaweera** [comment]
Kamal is a PHP web application framework, and its deployment and database choices depend on how you configure and design your application. Here are answers to your questions:

1. **Deploy Without a Database**: Yes, you can deploy a Kamal-based web application without a database if your application does not require one. Kamal does not force you to use a database, and you can create web applications that solely rely on static files or other data sources. You can deploy such applications as normal static websites.

2. **Database Flexibility**: Kamal itself does not lock you into a specific database system like MySQL. You have the flexibility to choose the database system that best fits your project's requirements. Kamal provides support for various database systems, including MySQL, SQLite, PostgreSQL, and others. You can configure your Kamal application to work with the database of your choice by changing the database settings in the configuration files.

   For example, if you want to use SQLite instead of MySQL, you would need to adjust the database configuration in your Kamal application to use SQLite as the database driver, provide the path to the SQLite database file, and set other relevant options.

   Here's a simplified example of how you might configure Kamal to use SQLite:

   ```php
   'database' => [
       'default' => 'sqlite',
       'connections' => [
           'sqlite' => [
               'driver'   => 'sqlite',
               'database' => __DIR__ . '/path/to/your/sqlite.db',
           ],
       ],
   ],
   ```

   In this configuration, the application uses SQLite as the database driver, and the database file is located at the specified path.

Remember that the actual steps and configuration may vary based on your specific Kamal application and deployment environment. Always refer to the Kamal documentation and database documentation for detailed setup instructions.
